lib/tasks/euros.rake

```diff
@@ -0,0 +1,128 @@
+        ]
+      },
+      'Group F' => {
+        api_id: nil,
+        api_code: 'GROUP_F',
+        teams: [
+          { name: 'Georgia', abbrev: 'GEO' },
+          { name: 'Portugal', abbrev: 'POR' },
+          { name: 'Czechia', abbrev: 'CZE' },
+          { name: 'Turkey', abbrev: 'TUR' }
+        ]
+      },
+    }
+    puts 'Creating the Euros...'
+    euros = Competition.find_or_create_by!(name: 'Euros 2024', start_date: Date.new(2024, 06, 14), end_date: Date.new(2024, 07, 14), api_id: 2018, api_code: 'EC')
+    puts '.. created the Euros'
+
+    puts 'Creating or finding first round...'
+    first_round = Round.find_or_create_by!(name: 'Group Stage', number: 1, competition: euros, api_name: 'GROUP_STAGE')
+    euros.update!(current_round: first_round)
+    Round.find_or_create_by!(name: 'Round of 16', number: 2, competition: euros, api_name: 'LAST_16')
+    Round.find_or_create_by!(name: 'Quarter-finals', number: 3, competition: euros, api_name: 'QUARTER_FINALS')
+    Round.find_or_create_by!(name: 'Semi-finals', number: 4, competition: euros, api_name: 'SEMI_FINALS')
+    # TODO: Doesn't look like the API has the 3rd place playoff
+    # Round.find_or_create_by!(name: 'Third Place', number: 5, competition: euros, api_name: '3PPO')
+    Round.find_or_create_by!(name: 'Final', number: 6, competition: euros, api_name: 'FINAL')
+
+    puts 'Creating or finding groups...'
+    groups.each_key do |group_name|
+      puts "...#{group_name}..."
+      group = Group.find_or_create_by!(name: group_name, round: first_round, api_id: groups[group_name][:api_id], api_code: groups[group_name][:api_code])
+      groups[group_name][:teams].each do |team_hash|
+        puts "Name: #{team_hash[:name]}, Abbrev: #{team_hash[:abbrev]}"
+        team = Team.find_by(abbrev: team_hash[:abbrev]) || Team.create!(team_hash)
+        Affiliation.find_or_create_by!(team: team, group: group)
+      end
+    end
+    # Calling the API to create the matches
+    MatchUpdateJob.perform_now(euros.id)
+
+    # TODO: this only works when there are matches so you'll see 1 for now
+    puts "...#{euros.rounds.count} Total Round"
+    puts "...#{euros.teams.count} Total Teams"
+    puts "...#{euros.groups.count} Total Groups"
+
+    doug = User.find_by(email: 'douglasmberkley@gmail.com') || User.create(email: 'douglasmberkley@gmail.com', password: ENV['ADMIN_PASSWORD'], admin: true)
+    trouni = User.find_by(email: 'trouni@gmail.com') || User.create(email: 'trouni@gmail.com', password: ENV['ADMIN_PASSWORD'], admin: true)
+    james = User.find_by(email: 'devereuxjj@gmail.com') || User.create(email: 'devereuxjj@gmail.com', password: ENV['ADMIN_PASSWORD'], admin: true)
+    renato = User.find_by(email: 'renatonato_jr@hotmail.com') || User.create(email: 'renatonato_jr@hotmail.com', password: ENV['ADMIN_PASSWORD'], admin: true)
+    caio = User.find_by(email: 'caio.santos@msn.com') || User.create(email: 'caio.santos@msn.com', password: ENV['ADMIN_PASSWORD'], admin: true)
```

Comment:
We probably did that before, but I guess it's not ideal to commit real emails on a public repo.

Don't think we've been using the [Rails credentials](https://edgeguides.rubyonrails.org/security.html#custom-credentials) for this repo, but we should probably start.

In any case, no need to fix this right away 👍

Reply:
ha yeah. i just copied what we had in the seeds but see what you mean

Reply:
we could move them to the .env file maybe

Reply:
Yeah the problem is that these are in the git commit history, so moving them to env or credentials now wouldn't fully resolve the issue. I'll create a separate issue.